tests/clar_libgit2.c

```diff
@@ -158,6 +158,18 @@ int cl_rename(const char *source, const char *dest)
 
 #endif
 
+int cl_setenv_printf(const char *name, const char *fmt, ...)
+{
+	va_list args;
+	git_buf buf = GIT_BUF_INIT;
+
+	va_start(args, fmt);
+	cl_git_pass(git_buf_vprintf(&buf, fmt, args));
+	va_end(args);
+
+	return cl_setenv(name, git_buf_cstr(&buf));
```

Comment:
I'm not sure that this helper method is necessary, because it makes things like the fact that it's leaking `buf` a bit more subtle.